# Quiz - Numpy & Pandas
Type:Practice Quiz

Attempts:0/1

Questions:10

Time:30m

Total Marks:10

Instructions:

- Please refer to the Dateset_ODI - Dateset_ODI.csv dataset to attempt the quiz. Questions from 5 to 10 require working on this dataset. Please make sure to import the dataset into your Jupyter notebook before starting the quiz.
- Only attempt the quiz when you are prepared and have enough time on your hand to finish it.
- The quiz once opened, must be completed within the time frame provided. You CANNOT start the quiz, leave it unattended for an extended period of time and come back later to finish.
- Ensure there is a proper internet connection while taking up the quiz. Any breakup in the connection will automatically submit your quiz.
- Re-attempts will be provided if the quiz gets submitted for any of the above-mentioned reasons.
- If you face any other technical issues on Olympus, please share the screenshot with your Program Manager so that the team can understand and resolve it on priority.


Question 1
1 Mark

What will be the output of the below code?

In [37]:
import numpy as np
a = np.array([1, 10, 2, 0])
np.power(a, 2)

array([  1, 100,   4,   0])

[ 1, 10, 2, 0]

- [ 1, 100, 4, 0]

[ 1, 1000, 8, 0]

None

Question 2
1 Mark

Consider the following code:

In [38]:
import numpy as np
a = np.array([['m', 'l'],['n', 'o']])
a[0,1]

'l'

What will be the value of a[0,1]?

- 'l'

'm'

'n'

'o'

Question 3
1 Mark

Consider the following code:

In [39]:
import numpy as np
array = np.arange(10) 
np.hsplit(array, 2)

[array([0, 1, 2, 3, 4]), array([5, 6, 7, 8, 9])]

Which of the following code would give the below output?

`[array([0, 1, 2, 3, 4]), array([5, 6, 7, 8, 9])]`

np.vsplit(array, 2)

- np.hsplit(array, 2)

np.hsplit(array)

np.hsplit(array, 4)

Question 4
1 Mark

What will be the output of the following code?

In [40]:
import numpy as np
array1 = np.array([1, 0, 0, 1, 0])
array2 = np.arange(6)
array1 + array2

ValueError: operands could not be broadcast together with shapes (5,) (6,) 

array([1, 1, 2, 4, 4])

- Error

array([2, 0, 0, 2, 0])

array([2, 0, 0, 2, 0, 0])

Question 5
1 Mark

State whether the following statement is True or False:

All the columns in the dataset ODI are of 'object' data type.


In [ ]:
import pandas as pd
df = pd.read_csv('./data/Dateset_ODI+-+Dateset_ODI.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3932 entries, 0 to 3931
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Scorecard   3932 non-null   object
 1   Team 1      3932 non-null   object
 2   Team 2      3932 non-null   object
 3   Winner      3932 non-null   object
 4   Margin      3753 non-null   object
 5   Ground      3932 non-null   object
 6   Match Date  3932 non-null   object
dtypes: object(7)
memory usage: 215.2+ KB


Answer choices
Select only one option

- True

False

Question 6
1 Mark

Consider the below statements and select the correct option.

1. There are no missing values in the dataset ODI
2. Only 2 columns - ' Margin' and 'Winner' have missing values
3. The 'Margin' column has 179 missing values

In [ ]:
def missing_check(df):
    total = df.isnull().sum().sort_values(ascending=False)   # total number of null values
    percent = (df.isnull().sum()/df.isnull().count()).sort_values(ascending=False)  # percentage of values that are null
    missing_data = pd.concat([total, percent], axis=1, keys=['Total', 'Percent'])  # putting the above two together
    return missing_data # return the dataframe
missing_check(df)

Total   Percent
Margin        179  0.045524
Scorecard       0  0.000000
Team 1          0  0.000000
Team 2          0  0.000000
Winner          0  0.000000
Ground          0  0.000000
Match Date      0  0.000000

Answer choices
Select only one option

2 and 3 are correct

Only 1 is correct

- Only 3 is correct

Only 2 is correct

Question 7
1 Mark
Which team had won the 50th ODI match?

In [ ]:
df.iloc[50]

Scorecard         ODI # 51
Team 1             England
Team 2            Pakistan
Winner             England
Margin             94 runs
Ground            The Oval
Match Date    May 26, 1978
Name: 50, dtype: object

Answer choices
Select only one option

- England

Pakistan

New Zealand

Australia

Question 8
1 Mark
How many ODI matches have been played at the 'Devonport' ground?

In [47]:
df[df['Ground']=='Devonport']

Scorecard   Team 1       Team 2   Winner   Margin     Ground   Match Date
429  ODI # 430  England  West Indies  England  29 runs  Devonport  Feb 3, 1987

5

9

1

4

Question 9
1 Mark
How many times the team Australia has won the ODI matches?

In [ ]:
(df['Winner']=='Australia').sum()

555

440

696

95

555

Question 10
1 Mark
How many times the second team, i.e., 'Team 2' has won the ODI match?

In [ ]:
(df['Team 2']==df['Winner']).sum()

1637

Answer choices
Select only one option

1598

1637

1789

1865